tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java

```diff
@@ -35,14 +51,16 @@
  *
  * The YARN logs for the test should be in {@code <TonY>/target/MiniTonY}.
  */
-public class TestTonyE2E {
+public class TestTonyE2E implements CallbackHandler, TaskUpdateListener {
```

Comment:
It doesn't seem clean to me for `TestTonyE2E` to directly implement these interfaces. I suggest creating a nested class that implements the callback interfaces.